pwiz_aux/msrc/utility/vendor_api/UIMF/Jamfile.jam

```diff
@@ -28,13 +28,17 @@ path-constant API_PATH : $(PWIZ_ROOT_PATH)/pwiz_aux/msrc/utility/vendor_api/UIMF
 
 
 if [ modules.peek : NT ]
-{
+{
+
+if [ path.exists $(API_PATH)\\UIMFLibrary.dll ]

```

Comment:
The --i-agree-to-the-vendor-licenses is used as a proxy for building with binary support. It doesn't need to be: there's a without-binary-msdata rule that could be used for it instead. But one or the other needs to be checked in the uimf-dll-location rule in Jamroot.jam, and the .hasDLLs value should be set based on the return value of that, not a simple path.exists check in the vendor_api/Jamfile.jam. Does that make sense?

Reply:
okay, so what I put in was a mix between what was there and what is used in the Thermo jamfile; I didn't have a file search set up properly, so I couldn't find the source of 'uimf-file-location'. So, now knowing that, I basically should just revert the jamfile changes I did make, and modify the uimf-dll-location rule in Jamroot.jam...

Reply:
also, there are currently checks for msvc version 9.0 or 9.0 express; should this be changed to just a check for toolset msvc?

Reply:
No, it's checking for those versions as being explicitly NOT supported. Is UIMF .NET 4 now? 4.what?

Reply:
Currently compiled as .NET 4.6.1 (The first UIMFLibrary.dll update I gave you back around June 2015 was already .NET 4.0, and I think the version already in use was as well).

So, as a remark: the Thermo jamfiles (I'm assuming due to the different APIs now in use for x86 vs. x64) do check for "--i-agree-to-the-vendor-licenses", rather than depending on jamroot.jam rules.

Reply:
Would you agree with me changing the lib rules for UIMF in vendor_api and vendor_reader to `include <conditional>@msvc-dotnet-requirement`, and removing the checks for msvc 9.0/9.0express?

Reply:
Yes, good idea.